src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameRequestStream.cs

```diff
@@ -180,13 +183,17 @@ private Task<int> ValidateState(CancellationToken cancellationToken)
                 case FrameStreamState.Open:
                     if (cancellationToken.IsCancellationRequested)
                     {
-                        return TaskUtilities.GetCancelledZeroTask(cancellationToken);
+                        return _error != null ?
+                            TaskUtilities.GetFaultedTask(_error, cancellationToken) :
+                            TaskUtilities.GetCancelledZeroTask(cancellationToken);
```

Comment:
This seems unnecessary since `_error` cannot be set when this `_state` is `FrameStreamState.Open`.